sdk/cosmos/azure-cosmos-spark_3-0_2-12/src/main/scala/com/azure/cosmos/spark/CosmosRowConverter.scala

```diff
@@ -224,6 +244,7 @@ private object CosmosRowConverter
             case textNode : TextNode =>
                 parseDateTimefromString(textNode.asText()) match {
                     case Some(odt) => Timestamp.valueOf(odt.toLocalDateTime)
+                    case None => null
```

Comment:
This is one question I had from Scala, wouldn't this fall through to the `Timestamp.valueOf(value.asText())`, that was my original intention

Reply:
The None would happen is textNode would be of type TextNode but asText returns None (kind of null). 

I added the case None code because of a warning that the match did not have axhaustive cases - basically because the compiler recognizes that asText could technically return None.

if it is guaranteed that TextNode.asText would never return None/null in this case I would just use parseDateTimefromString(textNode.asText.get()) there in stead of the inner match/case.

Makes sense?

Reply:
`parseDatetimeFromString` returns an Option (because the text might not be a datetime), in the case of None (not a datetime), opting for null makes sense, or throwing an exception saying the value does not represent a valid datetime. Weird I missed this compiler messages, maybe I have something turned off.

Reply:
Warnings are generated by ScalaStyle (not compiler - was unclear before). You can see it form the mvn output when running mvn clean compile from the sdk/cosmos folder

Reply:
And ACK - retruning an exception makes more sense - changed in latest iteration. Good catch!